### PR TITLE
feature: force train/validate split on datasets when building model

### PR DESCRIPTION
**Pull request by marcellodebernardi**
This PR improves the robustness and soundness of the model building procedure by making the following changes:

1. Enforce a train/validation set split on the data that is applied to all training runs, instead of letting the LLM code generation handle this inconsistently
2. Update the training code generation prompts to request the model to use training data and validation data differently
3. Fix a few miscellaneous bugs

Note that this does not yet include retraining the best model on `train` + `validation` sets, or evaluating the final model's performance on a holdout test set. These will be added in a separate PR.

### CODE REVIEW

**Comment by marcellodebernardi** [comment]
To the points raised by Copilot:

1. The 0% test split is there in advance of an upcoming PR that will actually use the holdout test set
2. The `self.filedir` that I removed was not being used anymore

File: smolmodels/templates/prompts/training/fix.jinja [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Fix the previous solution based on the following information.
```
Comment (vaibs-d) [inline smolmodels/templates/prompts/training/fix.jinja:1]: There's something very satisfying about seeing these in jinja templates instead of the code⭐

**Review by vaibs-d — APPROVED** [review]
Let's go! LGTM 💯
